### ISSUE TITLE
Receiving fault when connect to serial

### ISSUE THREAD

**Issue by nlpng** [bgkatz/motorcontrol#3]
Hello,

Many thanks for sharing this great project.
I just start to use stm32cubeide, and trying out the source code.
But I continue to receive the following message when connected through serial after flash the build.

```
FAULT
VDS_OCP
GDF
UVLO
OTSD
VDS_HA
VDS_LA
VDS_HB
VDS_LB
VDS_HC
VDS_LC
SA_OC
SB_OC
SC_OC
OTW
CPUV
VGS_HA
VGS_LA
VGS_HB
VGS_LB
VGS_HC
VGS_LC
```
What would be the possible cause of this message? Could you possibly share some hints to debug this?
Seems this is because my control board problem? failed to talk to DRV8323? I replicated the board design from the 3phase_inegrated [repo](https://github.com/bgkatz/3phase_integrated). 

Many thanks

**Comment by bgkatz** [comment]
That indicates a hardware issue.  Does the board work with the old mbed-based firmware?  What voltage are you powering the board with?

**Comment by nlpng** [comment]
Thanks for the reply, I think I need to look into the circuit board. Sorry I haven't got much experience with debugging hardware, but is there specific pins I could look into to check the board function correctly.

It is very smooth to build with stm32cubeide, I found it was quite difficult to use only compiler for the old mbed-based firmware, error when import the mbed-dev dependent library due to large size I guess. I am also trying with mbed studio with mbed-os 5.12.2, but still no success to flash the old firmware. 

I am using 24V 400W power supply to power the board.

**Comment by nlpng** [comment]
Ah, it seems that I already made a silly mistake by using the wrong gate driver IC, looking into the schematic of the 3phase_inegrated, it was `DRV8323RH`, which uses hardware interface, please correct me if I am wrong. it should be using `DRV8323RS` that is based on the SPI.

**Comment by bgkatz** [comment]
Yes, DRV8323RS.  The actual manufacturer part number in the BOM excel file is correct - the eagle schematic having RH instead of RS is an artifact of copy-pasting to make the eagle footprint for the RS version.  Sorry for the confusion.

**Comment by harshdobariya** [comment]
Hello people,
I am getting the similar fault problems. I have used DRV8323RS and checked the design thoroughly. The communication with DRV is successful. It is after the calibration that these fault raises. I have print_fault enabled in while loop and after calibration is done, these faults get printed. I am using the old embed firmware. (Compiling in Keil mbed studio online)
